src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -34,348 +44,445 @@
+    }
 
+    @DataBoundSetter
+    public void setExclusionPattern(String exclusionPattern) {
+        this.exclusionPattern = exclusionPattern;
+    }
 
+    /** End Setters for Workflow **/
 
-	public String getMaximumClassCoverage() {
-		return maximumClassCoverage;
-	}
+    protected static void saveCoverageReports(FilePath destFolder, FilePath sourceFolder) throws IOException, InterruptedException {
+        destFolder.mkdirs();
 
+        sourceFolder.copyRecursiveTo(destFolder);
+    }
 
-	public boolean isChangeBuildStatus() {
-		return changeBuildStatus;
-	}
-	
-    public boolean getChangeBuildStatus() {
-		return changeBuildStatus;
-	}
-	
-	protected static void saveCoverageReports(FilePath destFolder, FilePath sourceFolder) throws IOException, InterruptedException {
-		destFolder.mkdirs();
-		
-		sourceFolder.copyRecursiveTo(destFolder);
-	}
-	
-	protected static String resolveFilePaths(AbstractBuild<?, ?> build, BuildListener listener, String input) {
+    protected String resolveFilePaths(Run<?, ?> build, TaskListener listener, String input) {
         try {
-           
-        	return build.getEnvironment(listener).expand(input);
-            
+
+            return build.getEnvironment(listener).expand(input);
+
         } catch (Exception e) {
-            listener.getLogger().println("Failed to resolve parameters in string \""+
-            input+"\" due to following error:\n"+e.getMessage());
+            listener.getLogger().println("Failed to resolve parameters in string \"" +
+                    input + "\" due to following error:\n" + e.getMessage());
         }
         return input;
     }
-	
-	protected static FilePath[] resolveDirPaths(AbstractBuild<?, ?> build, BuildListener listener, final String input) {
-		//final PrintStream logger = listener.getLogger();
-		FilePath[] directoryPaths = null;
-		try {
-			directoryPaths = build.getWorkspace().act(new FilePath.FileCallable<FilePath[]>() 
-			{
-				static final long serialVersionUID = 1552178457453558870L;
-
-				public FilePath[] invoke(File f, VirtualChannel channel) throws IOException {
-					FilePath base = new FilePath(f);
-					ArrayList<FilePath> localDirectoryPaths= new ArrayList<FilePath>();
-					String[] includes = input.split(",");
-					DirectoryScanner ds = new DirectoryScanner();
-			        
-			        ds.setIncludes(includes);
-			        ds.setCaseSensitive(false);
-			        ds.setBasedir(f);
-			        ds.scan();
-			        String[] dirs = ds.getIncludedDirectories();
-			        
-			        for (String dir : dirs) {
+
+    protected static FilePath[] resolveDirPaths(FilePath workspace, TaskListener listener, final String input) {
+        //final PrintStream logger = listener.getLogger();
+        FilePath[] directoryPaths = null;
+        try {
+            directoryPaths = workspace.act(new FilePath.FileCallable<FilePath[]>()
+            {
+                public void checkRoles(RoleChecker checker) throws SecurityException {
```

Comment:
this is the wrong thing to do.  use the correct sub type - see the javadoc for [FileCallable](http://javadoc.jenkins-ci.org/hudson/FilePath.FileCallable.html)